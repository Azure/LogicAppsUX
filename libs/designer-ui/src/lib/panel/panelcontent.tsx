import type { PanelTab } from '.';
import React from 'react';

export interface PanelContentProps {
  selectedTab?: string;
  tabs: PanelTab[];
}

export const PanelContent = ({ tabs, selectedTab }: PanelContentProps): JSX.Element => {
  return (
    <div className="msla-content">
      {
        tabs.find((t) => {
          return t.name === selectedTab;
        })?.content
      }
    </div>
  );
};
