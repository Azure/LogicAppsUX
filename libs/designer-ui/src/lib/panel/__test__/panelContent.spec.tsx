import React from 'react';
import { PanelContent, type PanelContentProps } from '../panelcontent';
import * as ReactShallowRenderer from 'react-test-renderer/shallow';
import { describe, vi, beforeEach, afterEach, beforeAll, afterAll, it, test, expect } from 'vitest';
describe('ui/workflowparameters/workflowparameter', () => {
  let minimal: PanelContentProps, renderer: ReactShallowRenderer.ShallowRenderer;
  beforeEach(() => {
    minimal = {
      nodeId: '',
      tabs: [],
      trackEvent: vi.fn(),
      selectTab: vi.fn(),
    };
    renderer = ReactShallowRenderer.createRenderer();
  });

  afterEach(() => {
    renderer.unmount();
  });

  it('should construct.', () => {
    const panelContent = renderer.render(<PanelContent {...minimal} />);
    expect(panelContent).toMatchSnapshot();
  });

  it('should render menu items when passed tabs.', () => {
    // TODO: 13865562 When Tabs get setup
  });
});
