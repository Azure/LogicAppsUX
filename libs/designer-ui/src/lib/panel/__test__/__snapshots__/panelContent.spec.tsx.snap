// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ui/workflowparameters/workflowparameter should construct. 1`] = `
<div
  id="msla-node-details-panel-"
>
  <ForwardRef
    aria-label="more panels"
  >
    <TabList
      onTabSelect={[Function]}
      style={
        {
          "margin": "0px -12px",
        }
      }
    >
      <OverflowMenu
        tabs={[]}
      />
    </TabList>
  </ForwardRef>
  <div
    className="msla-panel-content-container"
  />
</div>
`;
